src/front-end/src/identification/Users/components/UserList.tsx

```diff
@@ -19,23 +26,21 @@ interface UserListProps {
 const UserList = (props: UserListProps) => {
   const { users, roles, filterText, setFilterText, page, setPage, totalPages } =
     props;
-  const [currentTimer, setCurrentTimer] = React.useState<NodeJS.Timeout | null>(
-    null
-  );
+  const [filteredUsers, setFilteredUsers] = React.useState<MockUser[]>([]);
 
-  const delayFilterChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    if (currentTimer) {
-      clearTimeout(currentTimer);
-    }
+  useEffect(() => {
+    const filtered = users.filter((user) => user.email.includes(filterText));
+    setFilteredUsers(filtered);
+  }, [users, filterText]);
 
-    const timer = setTimeout(() => {
+  const delayFilterChange = debounce(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
       setFilterText(event.target.value);
       setPage(1);
-      setCurrentTimer(null);
-    }, 666);
+    },
+    575
```

Comment:
Why 575, is this just recommended?

Reply:
I was experimenting with different durations and I arbitrarily left it with 575.